synthesizer/src/vm/finalize.rs

```diff
@@ -1258,6 +1258,22 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
 
                     // Insert the next committee into storage.
                     store.committee_store().insert(state.block_height(), next_committee)?;
+
+                    #[cfg(feature = "history")]
+                    {
+                        // Load a `History` object.
+                        let history = History::new(N::ID, store.storage_mode().clone());
+
+                        // Write the delegated mapping as JSON.
+                        history.store_mapping(state.block_height(), MappingName::Delegated, &next_delegated_map)?;
```

Comment:
Can we make a back-of-the-envelope calculation how much disk space the 3 mappings require? Imagine the worst case 100_000 delegators, wouldn't this already be *at least* 3MiB? And at 5 seconds per block, that would be 18 TiB per year? This number might be too optimistic (because JSON is bigger than 3MiB) or too pessimistic (because there won't be 100_000 delegators that quickly.

I think the only way to overcome this is to only store incremental data...

Reply:
We considered this in our design process.
An incremental data solution would have a smaller footprint, but require much more complicated indexing logic. 
In favor of simplicity, we leave management of the `history` directory to the node operators.